src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java

```diff
@@ -97,38 +103,49 @@ public void setup() {
         customShelfService.save(customShelf);
 
         Assumptions.assumeTrue(predefinedShelfService != null);
-        bookForm = createBookForm(true);
-
+        bookForm = createBookForm(READ, true);
     }
 
-    private BookForm createBookForm(boolean isInSeries) {
+    private BookForm createBookForm(ShelfName shelf, boolean isInSeries) {
         BookForm bookForm = new BookForm(predefinedShelfService, customShelfService);
-        bookForm.setBook(createBook(predefinedShelfService, isInSeries));
+        readShelf = predefinedShelfService.findAll().get(2);
```

Comment:
Could you change the assignment to be the result of our `findReadShelf()` method in `PredefinedShelfUtils`?

Reply:
That leads to the error "Non-static method 'findReadShelf()' cannot be referenced from a static context".
I could make the findReadShelf() method static, but that would mean that I would also have to make findPredefinedShelf() static and the PredefinedShelfService in PredefinedShelfUtils would also need to be static (and no longer final).

Maybe a simpler way which is cleaner that my previous approach would be to use
```java
readShelf = predefinedShelfService.findAll(READ).get(0);
```

Reply:
I agree that your new way is cleaner, but you should be able to access `findReadShelf()` since you're not accessing it from a static method. You just need to create an instance of the class first.

```java
PredefinedShelfUtils predefinedShelfUtils = new PredefinedShelfUtils(predefinedShelfService);
predefinedShelfUtils.findReadShelf();
```

I'd recommend making `predefinedShelfUtils` global and assign the value in `setup()`. You should then be able to access it where needed.

Perhaps you tried accessing `findReadShelf()` via the class (`PredefinedShelfUtils.findReadShelf`), not an instance?

Reply:
Oh god... i am such an idiot. ._. Now it is working.

Reply:
Glad to hear that you got it work